app/data_client.go

```diff
@@ -238,8 +246,25 @@ func captureMetadataFromProto(proto *pb.CaptureMetadata) CaptureMetadata {
 	}
 }
 
+func convertMapToProtoAny(input map[string]interface{}) (map[string]*anypb.Any, error) {
+	protoMap := make(map[string]*anypb.Any)
+	for key, value := range input {
+		structValue, err := structpb.NewValue(value)
+		if err != nil {
+			return nil, err
+		}
+		anyValue, err := anypb.New(structValue)
+		if err != nil {
+			return nil, err
+		}
+		protoMap[key] = anyValue
+	}
+	return protoMap, nil
+}
+
```

Comment:
Can we put this in `protoutils`? It's a pretty generic function, and while `data_client` is the only current user, I suspect there may be broader use in the future (much like `ConvertStringMapToAnyPBMap`).